File: Codingt/src/Programmers/lv01/KthNumber.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

int cuttedNumber(vector<int> arr, vector<int> command);
vector<int> cuttedArray(vector<int> arr, vector<vector<int>> commands);
vector<int> sortArray(vector<int> arr);

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer = cuttedArray(array, commands);
    return answer;
}

vector<int> cuttedArray(vector<int> arr, vector<vector<int>> commands){
    vector<int> cuttedArr;
    for(int i =0;i<commands.size();i++){
        cuttedArr.push_back(cuttedNumber(arr,commands[i]));
    }
    return cuttedArr;
}
int cuttedNumber(vector<int> arr, vector<int> command){
    vector<int> unsortedArr, sortedArr;
    for(int i = command[0]-1;i < command[1]; i++){
        unsortedArr.push_back(arr[i]);
    }
    sortedArr = sortArray(unsortedArr);
    return sortedArr[command[2]-1];
}
vector<int> sortArray(vector<int> arr){
    sort(arr.begin(), arr.end());
    return arr;
}

File: Codingt/src/Programmers/lv01/KthNumber.java
package Programmers.lv01;

import java.util.Arrays;

public class KthNumber {
	public static void main(String[] args) {
		KMSolution sol = new KMSolution();
	}
}

class KMSolution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = cuttedArray(array, commands);
        return answer;
    }
    int[] cuttedArray(int[] arr, int[][] commands){
        int[] cuttedArr = new int[commands.length];
        for(int i =0; i<commands.length;i++){
            cuttedArr[i] = cuttedNumber(arr, commands[i]);
        }
        return cuttedArr;
    }
    int cuttedNumber(int[] arr, int[] command){
        int[] numArr = new int[command[1]-command[0]+1];
        int last=0;
        for(int i = command[0]-1;i < command[1]; i++){
            numArr[last++] = arr[i];
        }
        Arrays.sort(numArr);
        return numArr[command[2]-1];
    }
}